projects/pgai/pgai/vectorizer/embeddings.py

```diff
@@ -165,6 +165,17 @@ async def setup(self) -> None:  # noqa: B027 empty on purpose
         """
 
 
+class BaseURLMixin:
```

Comment:
I prefer to always provide constructor to classes to avoid inconsistent state between instantiation and usage of classes, but this is just MHO and more about OOP in general.

Reply:
Mixed feelings here. Simplicity applies here IMHO: not having to deal with calls to sup classes from child classes gives me confidence on this. Otherwise, I would even opt for not creating sub a mixin class. However, I wanted to start having a class to be reused by any other embedded that can set their base_url, so if needed, we can know if a class implements that.

Having said that, I'm a completely open to any recommendation from any python expert here as I do not have enough experience in this language unfortunately. I always advocate for creating idiomatic code rather than focus on general practices because Developer Experience is what matters the most here.